secboot/secboot_tpm.go

```diff
@@ -382,6 +386,49 @@ func ResealKeys(params *ResealKeysParams) error {
 	return sbUpdateKeyPCRProtectionPolicyMultiple(tpm, params.KeyFiles, authKey, pcrProfile)
 }
 
+func TpmPrepare(dir string) error {
+	tpm, err := insecureConnectToTPM()
+	if err != nil {
+		if xerrors.Is(err, sb.ErrNoTPM2Device) {
+			return nil
+		}
+		return fmt.Errorf("cannot open TPM connection: %v", err)
+	}
+	defer tpm.Close()
+
+	if !isTPMEnabled(tpm) {
+		return nil
+	}
+
+	srkTmplPath := filepath.Join(dir, "tpm2-srk.tmpl")
```

Comment:
I'm not sure what this and the code below is meant to do/ how the overall system is designed, so I will leave out this part from my review, but someone else should review this part for correctness, I don't see any logic errors like not handling an error anything but that isn't good enough :smile:

Reply:
Take a blob. Offer the blob to the TPM. Cross fingers and hope that TPM will nom the blob. And thus will allow to unseal secrets that have been presealed for this TPM using this template.